src/pages/BooksList.jsx

```diff
@@ -53,22 +54,52 @@ function BooksList() {
   if (loading) {
     return <p>Loading...</p>;
   }
+  const subjects = [
+    ...new Set(
+      booksData.flatMap((book) => book.subjects.map((subject) => subject.name))
+    ),
+  ];
+
+  const filteredBooks = booksData.filter((book) => {
```

Comment:
Here, if you want to avoid recreating the `filteredBooks` array on every render, you can use a `useMemo`. We usually use it more in mobile dev than in web dev, but it's always interesting to read the docs.

Reply:
I will improve my skills and read the doc about this useMemo thanks!